node/subsystem-util/src/runtime/mod.rs

```diff
@@ -109,8 +109,14 @@ impl RuntimeInfo {
 	/// Create with more elaborate configuration options.
 	pub fn new_with_config(cfg: Config) -> Self {
 		Self {
-			session_index_cache: LruCache::new(max(10, cfg.session_cache_lru_size)),
-			session_info_cache: LruCache::new(cfg.session_cache_lru_size),
+			session_index_cache: LruCache::new(
+				NonZeroUsize::new(max(10, cfg.session_cache_lru_size))
+					.expect("Session index cache size should not be 0."),
+			),
+			session_info_cache: LruCache::new(
+				NonZeroUsize::new(cfg.session_cache_lru_size)
+					.expect("Session info cache size should not be 0."),
+			),
```

Comment:
```suggestion
			session_info_cache: LruCache::new(NonZeroUsize::new_unchecked(max(10, cfg.session_cache_lru_size))),
```

Or probably better to make `session_cache_lru_size` `NonZeroUsize`.